packages/material-ui-styles/src/ThemeProvider/ThemeProvider.js

```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { exactProp } from '@material-ui/utils';
+import { CacheProvider } from '@emotion/core';
+import createCache from '@emotion/cache';
+import rtlPlugin from 'stylis-plugin-rtl';
```

Comment:
It is removed from the component and used in the `ThemeProvider` now. If you think it would be problem to have it in the `ThemeProvider` too, we could create a package for `RtlProvider` that can set the caching.

Reply:
Based on the popularity of RTL: https://npm-stat.com/charts.html?package=jss&package=jss-rtl, we came to the conclusion that the safest approach is to start with a solution based on the documentation.

However, we will likely need to look at the cases where RTL needs to be disabled. We have cases like this in the codebase with `flip: false`.